blockchain/validators/src/metadata/key_range/config.rs

```diff
@@ -0,0 +1,442 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Configuration for the key range validator.
+
+use crate::ParseError;
+use hex::ToHex;
+use mc_blockchain_types::BlockIndex;
+use mc_crypto_keys::{DistinguishedEncoding, Ed25519Public};
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::BTreeMap,
+    ffi::OsStr,
+    fs,
+    ops::RangeInclusive,
+    path::{Path, PathBuf},
+};
+
+/// Container for keys with a range of block indexes that are valid per key.
+pub type KeyValidityMap = BTreeMap<Ed25519Public, Vec<RangeInclusive<BlockIndex>>>;
+
+const PUBLIC_KEY_TAG: &str = "PUBLIC KEY";
+
+/// Container for key validity configs.
+/// Supports parsing a `metadata-signers.toml` as specified in [MCIP #43](https://github.com/mobilecoinfoundation/mcips/pull/43).
+#[derive(Clone, Debug, Deserialize, Eq, Ord, PartialEq, PartialOrd)]
+pub struct Config {
+    /// The key validity ranges.
+    #[serde(alias = "node")] // alias for parsing
+    pub configs: Vec<KeyValidity>,
+
+    /// Optional base path for parsing relative PEM file paths.
+    #[serde(skip)]
+    pub base_path: Option<PathBuf>,
+}
+
+impl Config {
+    /// Load the config as TOML or JSON from the given file path.
+    pub fn load(path: impl AsRef<Path>) -> Result<Self, ParseError> {
+        let path = path.as_ref();
+        let bytes = fs::read(path)?;
+        let mut config: Config = match path.extension().and_then(OsStr::to_str) {
+            Some("toml") => Ok(toml::from_slice(&bytes)?),
+            Some("json") => Ok(serde_json::from_slice(&bytes)?),
+            _ => Err(ParseError::UnrecognizedExtension(path.into())),
+        }?;
+        config.base_path = path.parent().map(Into::into);
+        Ok(config)
+    }
+
+    /// Get a [KeyValidityMap] from this config.
+    pub fn to_validity_map(&self) -> Result<KeyValidityMap, ParseError> {
```

Comment:
This `struct Config` is the piece that leverages automatic serde, with a structure matching the config file format, and this method maps to the internal representation.

Reply:
I initially had `parse_pem_or_hex` in a `serde` annotation, but I moved it to this method so we can treat the relative paths  as relative to the TOML file rather than CWD.